lib/spack/spack/cmd/__init__.py

```diff
@@ -381,25 +381,27 @@ def _key(item):
         return str(item)
 
     # Traverse the index and print out each package
-    for i, (architecture, compiler) in enumerate(sorted(index, key=_key)):
+    for i, (architecture, compilers) in enumerate(sorted(index, key=_key)):
         if i > 0:
             print()
 
+        # Drop the leading space from compilers to clean up output and aid checks.
+        compilers_info = compilers[1:] if compilers.startswith(" ") else compilers
```

Comment:
```suggestion
        compilers_info = compilers.strip() or “no compilers”
```

or `lstrip()` if it’s really only needed on the left, but strip seems fine.

Reply:
🤦